airflow/models/dag.py

```diff
@@ -179,6 +179,20 @@ def get_last_dagrun(dag_id, session, include_externally_triggered=False):
     return query.first()
 
 
+def get_first_dagrun(dag_id, session, include_externally_triggered=False):
+    """
+    Returns the first dag run for a dag, None if there was none.
+    First dag run can be any type of run eg. scheduled or backfilled.
+    Overridden DagRuns are ignored.
+    """
+    DR = DagRun
+    query = session.query(DR).filter(DR.dag_id == dag_id)
+    if not include_externally_triggered:
+        query = query.filter(DR.external_trigger == expression.false())
+    query = query.order_by(DR.execution_date.asc())
+    return query.first()
```

Comment:
Do we _need_ this as a separate function? (I don’t like `get_last_dagrun` to begin with and would prefer we not adding more things like it unless needed.)